runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/grpcclient/GrpcWindmillServer.java

```diff
@@ -294,7 +293,10 @@ private synchronized CloudWindmillServiceV1Alpha1Grpc.CloudWindmillServiceV1Alph
     if (stubList.size() == 1) {
       return stubList.get(0);
     }
-    return stubList.get(rand.nextInt(stubList.size()));
+    return stubList
+        .get(rand.nextInt(stubList.size()))
+        .withDeadlineAfter(
+            AbstractWindmillStream.DEFAULT_STREAM_RPC_DEADLINE_SECONDS, TimeUnit.SECONDS);
```

Comment:
this deadline is not being applied to stub returned above if only 1 stub

Reply:
done